serde_json/src/de.rs

```diff
@@ -145,95 +151,157 @@ impl<Iter> Deserializer<Iter>
         Ok(())
     }
 
-    fn parse_number<V>(&mut self, mut visitor: V) -> Result<V::Value, Error>
+    fn parse_integer<V>(&mut self, pos: bool, visitor: V) -> Result<V::Value>
         where V: de::Visitor,
     {
-        let mut neg = false;
-
-        if self.ch_is(b'-') {
-            try!(self.bump());
-            neg = true;
-        }
-
-        let res = try!(self.parse_integer());
+        match self.ch_or_null() {
+            b'0' => {
+                try!(self.bump());
 
-        if self.ch_is(b'.') || self.ch_is(b'e') || self.ch_is(b'E') {
-            let mut res = res as f64;
+                // There can be only one leading '0'.
```

Comment:
Yeah, this is just from the [json](http://json.org/) grammar. serde_json strictly implements the spec, so it has to reject numbers like `00`.